Cython/Utils.py

```diff
@@ -316,7 +316,10 @@ def find_versioned_file(directory, filename, suffix,
     assert not suffix or suffix[:1] == '.'
     path_prefix = os.path.join(directory, filename)
 
-    matching_files = glob.glob(path_prefix + ".cython-*" + suffix)
+    matching_files = glob.glob(
+        (glob.escape(path_prefix) if sys.version_info >= (3, 4) else
+            ''.join([ '['+c+']' if c in '[]*?' else c for c in path_prefix]))
```

Comment:
I don't think generating `[]]` for `]` helps anyone. We'd better leave it as it is, i.e. exclude it from escaping. That's also [what CPython does](https://github.com/python/cpython/blob/cbe809dfa94385bacaa24beee3976ba77a395589/Lib/glob.py#L220).
```suggestion
            ''.join([ '['+c+']' if c in '[*?' else c for c in path_prefix]))
```